src/root.zig

```diff
@@ -198,6 +224,10 @@ pub fn AlgorithmType(
             const rows = needle.len;
             const cols = haystack.len;
 
+            // TODO: resize if needed instead of reallocate
+            self.deallocateMatrixAndBuffer();
+            self.allocateMatrixAndBuffer(cols, rows) catch @panic("Memory error");
```

Comment:
Let's not do this in `score`. Provide instead a `resize` function that the user can call to resize the matrices to a new maximum size, and use a `std.debug.assert` here to check the sizes. I don't want to make the `score` function in Algorithm throw errors from doing memory things, as that would break the current API in an unnecessary way.

Reply:
Okay, I will make changes in #5

Reply:
Just a little problem in Unicode with the `score` function : during string conversion, error can be throws. What do we do ? Letting `score` throwing error or use `@panic` ?